posthog/api/event.py

```diff
@@ -159,22 +190,35 @@ def list(self, request: request.Request, *args: Any, **kwargs: Any) -> response.
             # Result with selected columns
             if isinstance(query_result, EventsQueryResponse):
                 return response.Response(
-                    {
-                        "columns": select,
-                        "types": query_result.types,
-                        "results": query_result.results,
-                        "hasMore": query_result.has_more,
-                    },
+                    {"columns": select, "types": query_result.types, "results": query_result.results}
                 )
 
             result = ClickhouseEventSerializer(
                 query_result[0:limit], many=True, context={"people": self._get_people(query_result, team)}
             ).data
 
             next_url: Optional[str] = None
-            if not is_csv_request and len(query_result) > limit:
-                next_url = self._build_next_url(request, query_result[limit - 1]["timestamp"], order_by)
-            return response.Response({"next": next_url, "results": result})
+            previous_url: Optional[str] = None
+            is_request_paginated = request.GET.get("after") or request.GET.get("before")
+            if not is_csv_request and len(query_result) > 0:
+                if len(query_result) > limit:
+                    next_url = self._build_pagination_url(
+                        request, "next", query_result[limit - 1]["timestamp"], order_by
+                    )
+
+                if is_request_paginated:
+                    previous_url = self._build_pagination_url(
+                        request, "previous", query_result[0]["timestamp"], order_by
+                    )
+
+            return response.Response(
+                {
+                    "next": next_url,
+                    "previous": previous_url,
+                    "results": result,
+                    "count": len(result),
```

Comment:
This'll be wrong as the result is limited. We also in general do not want to return a count for events as that'd mean getting _all_ events to calculate just this property.

Reply:
Hello @timgl!

Maybe you make separated method for getting count of events? It's important thing for some cases.

For example: we use PostHog as a main analytics tool and would like to make Results of Year for our users, but without `count` property we can't calculate events quickly and have big problems.

Reply:
Hey @hit2hat the separate method you want to use here to get total counts is the `trends` API, which does this aggregation over a time range. (it's the same as what you see for the graphs on the trends insight inside PostHog)

Reply:
Hello!

Seems like a solution, thanks. But it's may be not clear for other PostHog users.

Reply:
@timgl maybe I misunderstand how querying clickhouse works, but isn't the count I'm returning here just the length of the limited query result (see line 196-198), which we are returning anyways?

Reply:
Yes, @kappa90 , but as you say, it's not very useful if it's only the limited count (you already can compute this data) & not the total count of events in the time range.

In such an API, the count is expected to return the total, not the count after applying the limits.